### PR TITLE
Updated reflection logic to find cleaner in new place

### PR DESCRIPTION
**Pull request by DavyLandman**
This gets rid of the warning in issue #119 but migrating to java modules in the future would still be prefered, as that would drop the requirement for reflection

### CODE REVIEW

**Comment by jurgenvinju** [comment]
Ok, that's fine. Why do modules remove the need for reflection in this case?

**Comment by jurgenvinju** [comment]
CF complains:

```
rror: ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project vallang: Compilation failure
Error: ERROR] /home/runner/work/vallang/vallang/src/main/java/io/usethesource/vallang/io/binary/util/FileChannelDirectInputStream.java:[91,46] error: [argument] incompatible argument for parameter obj of get.
Error: ERROR]   found   : null (NullType)
Error: ERROR]   required: @Initialized @NonNull Object
```

**Comment by DavyLandman** [comment]
> Ok, that's fine. Why do modules remove the need for reflection in this case?

The full explanation is here: https://github.com/usethesource/vallang/issues/136#issuecomment-1002634668 Roughly, we get access to the internal class that we need that way.

**Comment by DavyLandman** [comment]
> CF complains:
> 
> ```
> rror: ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project vallang: Compilation failure
> Error: ERROR] /home/runner/work/vallang/vallang/src/main/java/io/usethesource/vallang/io/binary/util/FileChannelDirectInputStream.java:[91,46] error: [argument] incompatible argument for parameter obj of get.
> Error: ERROR]   found   : null (NullType)
> Error: ERROR]   required: @Initialized @NonNull Object
> ```

interesting, that looks like an error in CF description for the reflection api, as that api allows a null in case of a static field. I'll see if I can work around it in a different way.

**Comment by DavyLandman** [comment]
This CF issue discusses this get problem, but it's a bit unclear what is happening here: https://github.com/typetools/checker-framework/issues/3223

**Comment by DavyLandman** [comment]
> > Ok, that's fine. Why do modules remove the need for reflection in this case?
> 
> The full explanation is here: [#136 (comment)](https://github.com/usethesource/vallang/issues/136#issuecomment-1002634668) Roughly, we get access to the internal class that we need that way.

Let me rephrase this a bit, we need reflection to get access to the unsafe class (that's the contract, you cannot statically compile against that class). In jdk9+ it looks like you can get class-loader access to the unsafe class  via `jdk.unsupported` (see: https://bugs.openjdk.java.net/browse/JDK-8153737). So that would drop this reflection party, and we could invoke functions on the Unsafe class as we would other code.
